### PR TITLE
Add overview page about TerraModel to docs and build examples

### PR DESCRIPTION
**Pull request by anowacki**
This is a stab at an overview page for `TerraModel` which introduces some of the
conventions and lists some of the functionality.  If there are other things which 
should go in, it would be useful to know.

Along the way, it struck me that it would be easiest to explain the different 
coordinate systems, so I have made a page for that.

I don't know if SVG images can be included, but I have tried that.

- Add an overview page for `TerraModel`s which refers to the
  autogenerated docstrings.
- Add a page about the reference frames used within terratools
  with an SVG figure.

- JOSS: Fix typo in paper manuscript
- Docs: Add `TerraModel` overview and reference frame page
- Fix reference to MIT licence and copyright dates

Edit: This PR has also grown to include actually including the example scripts and fixing breakages from them.

### CODE REVIEW

**Comment by andreww** [comment]
Do we get previews of updated documentation on read the docs? If not, can we turn this on (apparently there is a 'build pull requests for this project' as an advanced setting on readthedocs)?

I see readthedocs is currently failing its build - since https://github.com/mantle-convection-constrained/terratools/commit/fb8487aa531e67933dc9179b2bfe600ccaef71fa - not sure what's going on.

**Comment by anowacki** [comment]
It would be nice to have PR docs builds.  @bobmyhill are you able to help here?  You nicely set all the doc building up for us.

**Comment by anowacki** [comment]
> see readthedocs is currently failing its build - since [fb8487a](https://github.com/mantle-convection-constrained/terratools/commit/fb8487aa531e67933dc9179b2bfe600ccaef71fa) - not sure what's going on.

Hm, that's surprising.  Docs building should pass for CI as a whole to pass.

**Comment by anowacki** [comment]
Docs build failure log [here](https://readthedocs.org/projects/terratools/builds/21812769/).

It looks a bit like an API change has bitten us upon an upgrade or something.

(Note: to see docs builds, go to the docs page on readthedocs, click on the 'v:latest' button at the bottom right, then 'Builds' in an option to click.)

**Comment by anowacki** [comment]
Moved issue around docs build failure to #120.

**Comment by anowacki** [comment]
Docs building on PR moved to #121

**Comment by anowacki** [comment]
@andreww the build for this PR is at https://terratools--112.org.readthedocs.build/en/112/.

**Review by andreww — APPROVED** [review]
A couple of minor things here - don't know if you want to fix them in this PR or not. Anyway, I'm relaxed about merging.

File: docs/model.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+which represents an instance in time within a TERRA simulation.
+
+## `terratools.terra_model`
+
+The `terra_model` module within `terratools` contains all the functionality
+related to using `TerraModel`s.  Load it with
+`from terratools import terra_model`.
+
+## Reading NetCDF files
+`terratools` defines a NetCDF file format [REF TO FILE FORMAT], which can be
```
[the remark this answers is not in the document]
Comment (anowacki) [reply docs/model.md:11]: Thanks—done.

File: docs/model.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  [terra_model.TerraModel.get_radii][terratools.terra_model.TerraModel.get_radii].
+  For a model `m`, the `i`th layer is therfore at radius `m.get_radii()[i]`.
+- The second index gives the lateral
+  ([global geographic](reference_frames#Global-geographic)) position on the unit
+  sphere of the point.
+  [terra_model.TerraModel.get_lateral_points][terratools.terra_model.TerraModel.get_lateral_points]
+  returns a tuple of `lon, lat`, where `lon` and `lat` are both the global
+  geographic coordinates of each lateral point.  Therefore the coordinates of
+  the `j`th lateral point for model `m` are given by
+  ```python
```
[the remark this answers is not in the document]
Comment (anowacki) [reply docs/model.md:63]: Hopefully fixed.

**Pull request merged**